### PR TITLE
Add getters to FaciaContent

### PR DESCRIPTION
**Pull request by DiegoVazquezNanini**
I added a bunch of getters to the FaciaContent  to avoid using the trait and to Allow Francis to refactor this file in the future.

### CODE REVIEW

**Comment by adamnfish** [comment]
There are no tests!

We don't need to test every detail, but stuff where there's logic si worth testing (e.g. falling back to `"Missing Headline"` - since that String is hard coded clients might implement against it and the test will stop changes breaking things)

File: fapi-client/src/test/scala/com/gu/facia/api/models/FaciaContentHelperTest.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    val snap = LatestSnap("myId", None, None, None, None, None, None, "myGroup", None, false, false, false, false, false, false, None, false, None, None, false, false)
+    FaciaContent.headline(snap) should equal("Missing Headline")
+  }
+
+  "should return the headline for a CuratedContent" in {
+    val cc = CuratedContent(null, Nil, "The headline", None, None, "myGroup", None, false, false, false, false, false, None, false, None, None, false, false)
+    FaciaContent.headline(cc) should equal("The headline")
+  }
+
+  //href
```
Comment (adamnfish) [inline fapi-client/src/test/scala/com/gu/facia/api/models/FaciaContentHelperTest.scala]: what's this comment here for?

File: fapi-client/src/test/scala/com/gu/facia/api/models/FaciaContentHelperTest.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.gu.facia.api.models
+
+import com.gu.contentapi.client.model.Content
+import org.scalatest.{FreeSpec, Matchers}
+
+class FaciaContentHelperTest extends FreeSpec with Matchers {
+
+  //headline
```
Comment (adamnfish) [inline fapi-client/src/test/scala/com/gu/facia/api/models/FaciaContentHelperTest.scala]: what's this comment here for?

**Comment by adamnfish** [comment]
Nice to have those tests, also interesting that adding tests exposed that the other way of implementing the methods was less testable!

Not sure why there are a few `null`s in the tests though? Is there not a sensible default value? If you find yourself using nulls in tests to get around having to make concrete instances of things then consider using mocks. They're nulls as well, but without all the boiler plate. In these tests mocks might make it much clearer because you'd be doing the setup the test needs explicitly, rather than having the important fields hidden in with all the default values (`false`, `None`, etc). Up to you!

**Comment by DiegoVazquezNanini** [comment]
@adamnfish the last commit address all your comment, except the suggestion of using mocks.

**Comment by adamnfish** [comment]
Looks fine to me, but be nice to get a thumbs up from the fronts team because I can't speak for the correctness of that logic :-/

**Comment by DiegoVazquezNanini** [comment]
@johnduffell @robertberry any comments ?

File: fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala [hunk 3]
```diff
@@ -55,6 +55,153 @@ object ImageCutout {
...
+    linkSnap => linkSnap.id,
+    latestSnap => latestSnap.id)
+
+  def content(fc: FaciaContent): Option[Content] = fold(fc)(
+    curatedContent => Option(curatedContent.content),
+    supportingCuratedContent => Option(supportingCuratedContent.content),
+    linkSnap => None,
+    latestSnap => latestSnap.latestContent)
+
+  def maybeContent(fc: FaciaContent) = content(fc)
```
Comment (robertberry) [inline fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala]: Why this alias?

File: fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala [hunk 4]
```diff
@@ -55,6 +55,153 @@ object ImageCutout {
...
+  def supportingContent(fc: FaciaContent): List[FaciaContent] = fold(fc)(
+    curatedContent => curatedContent.supportingContent,
+    supportingCuratedContent => Nil,
+    linkSnap => Nil,
+    latestSnap => Nil)
+
+  def headline(fc: FaciaContent): String = fold(fc)(
+    curatedContent => curatedContent.headline,
+    supportingCuratedContent => supportingCuratedContent.headline,
+    linkSnap => linkSnap.headline.getOrElse("Missing Headline"),
```
Comment (robertberry) [inline fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala]: Why not an Option?

(I'd tend to push rendering something else to the most outer layer possible to give consumers of the client the choice.)

File: fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala [hunk 5]
```diff
@@ -55,6 +55,153 @@ object ImageCutout {
...
+    latestSnap => latestSnap.headline.orElse(latestSnap.latestContent.map(_.webTitle)).getOrElse("Missing Headline"))
+
+  def href(fc: FaciaContent): String = fold(fc)(
+    curatedContent => curatedContent.href.getOrElse("Missing href"),
+    supportingCuratedContent => supportingCuratedContent.href.getOrElse("Missing href"),
```
Comment (robertberry) [inline fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala]: Ditto to above.

**Comment by robertberry** [comment]
Neat. I like the fold. :+1:

File: fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala (hunk 3 above)
Comment (DiegoVazquezNanini) [inline fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala]: @janua likes the maybe* for Options, but the actual name of the attribute is content so I guess I add that alias to make @janua happy.

Comment (robertberry) [inline fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala]: *shrugs* Seems kind of redundant when it's in the type signature (which always deserves more attention than the name!) but I'm not bothered if it's a standard that's already been set in this project.

File: fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala (hunk 4 above)
Comment (DiegoVazquezNanini) [inline fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala]: Yes, that make a lot of sense. Let me fix it.

**Comment by DiegoVazquezNanini** [comment]
Thansk @robertberry and @adamnfish

File: fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala [hunk 6]
```diff
@@ -55,6 +55,151 @@ object ImageCutout {
 
 sealed trait FaciaContent
 
+object FaciaContent {
+
+  def fold[T](fc: FaciaContent)(c: (CuratedContent) => T, scc: (SupportingCuratedContent) => T,
+              ls: (LinkSnap) => T, las: (LatestSnap) => T): T = fc match {
```
Comment (johnduffell) [inline fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala]: I know it's already merged, but I'd recommend giving things more descriptive names where possible.  We usually have autocomplete anyway, and it helps my poor brain =)

**Comment by DiegoVazquezNanini** [comment]
That is a fair comment. I will follow your advice in the future. Thanks @johnduffell :)

**Pull request merged**